server/src/server.js

```diff
@@ -25,12 +25,24 @@ app.use(checkJwt);
 // set up the api as middleware
 app.use('/api/v1', api);
 
+// handle all incorrect routes under `/api`
+app.all('*', (req, res, next) => {
+  const err = new Error();
+  err.name = 'RouteNotFoundError';
```

Comment:
Any reason not to just do
```javascript
const err = new Error('RouteNotFoundError')
```
that way the description will stay in `message`? I can see some kind of enhanced `Error` class being useful, so you can track `statusCode` and `name` and `message` and `stack` or something, but for now maybe just using the default `Error` message field would be fine

Reply:
+ should `RouteNotFoundError` go in `stringConstants`?

Reply:
Actually in the test file you already have `strings.errors.NONEXISTENT` - is that the same jawn?

Reply:
The reasoning is to follow the convention of errors thrown by 3rd party packages (errors I have no control over). They all set the `name` property of the error. Calling
```js
const err = new Error('RouteNotFoundError')
```
sets the `message` property. To be as abstract as possible, the error handler is switching on the `name` property because that has shown to be a good way to distinguish what kind of error is being thrown by the third party packages.

As for the error constants, not the same. I don't think there's any reason to put it in the constants file because most of the names only get used once (for now). Yes, that name appears twice in the code but I think that will not occur too often. If it does I can move it into constants.

Reply:
I see now, cool beans. I think I'm going to implement custom errors on the front-end end at some point, as it'll help for displaying user friendly messages / logging. Might be useful on the back-end at some point too. Something like [this]():

```javascript
module.exports = class AppError extends Error {
  constructor (message, status) {
  
    // Calling parent constructor of base Error class.
    super(message);
    
    // Saving class name in the property of our custom error as a shortcut.
    this.name = this.constructor.name;

    // Capturing stack trace, excluding constructor call from it.
    Error.captureStackTrace(this, this.constructor);
    
    // You can use any additional properties you want.
    // I'm going to use preferred HTTP status for this error types.
    // `500` is the default value if not specified.
    this.status = status || 500;
    
  }
};
```
Maybe a future enhancement

Reply:
Ye, I was reading about that the other day and think it would definitely be beneficial in the future.